src/components/survey/generic/index.js

```diff
@@ -0,0 +1,128 @@
+import {faArrowLeft} from "@fortawesome/free-solid-svg-icons";
+import {useEffect, useState} from "react";
+import Icon from "components/common/icon";
+import Markdown from "components/common/markdown";
+import Modal from "components/common/modal";
+import useAssessment from "lib/hooks/use-assessment";
+import useGraphql from "lib/hooks/use-graphql";
+import useHttp from "lib/hooks/use-http";
+import useTranslate from "lib/hooks/use-translate";
+import Container from "./container";
+import QuestionSet from "./question-set";
+import style from "./style.scss";
+
+export default function Generic() {
+  const [questionSetIndex, setQuestionSetIndex] = useState(0);
+  const [answers, setAnswers] = useState([]);
+  const [showInstructions, setShowInstructions] = useState(false);
+  const [showConclusions, setShowConclusions] = useState(false);
+
+  const assessment = useAssessment({surveyType: "generic"});
+  const questionSets = assessment ? assessment.survey.questionSets : [];
+  const questionCount = questionSets.reduce((count, set) => count + set.questions.length, 0);
+  const currentQuestionSet = questionSets ? questionSets[questionSetIndex] : {};
+  const progress = questionSetIndex >= 0 ? (questionSetIndex / questionSets.length) * 100 : 0;
+  const finished = questionSets.length > 0 && questionCount === answers.length;
+
+  const graphQL = useGraphql();
+  const http = useHttp();
+  const translate = useTranslate();
+  const props = {progress};
+
+  const updateAnswer = (questionId, selectedOptionId) => {
+    const currentAnswers = answers.filter((answer) => answer.questionId !== questionId);
+    setAnswers([...currentAnswers,
+      {questionId, selectedResponseOptionId: selectedOptionId}]);
+  };
+
+  const next = () => { setQuestionSetIndex(questionSetIndex + 1); };
+  const back = () => { setQuestionSetIndex(questionSetIndex - 1); };
+
+  const onSubmit = () => {
+    const query = graphQL.generic.update;
+    const variables = {
+      assessmentID: assessment.id,
+      answers
+    };
+
+    http.post(graphQL.generic.path, {query, variables}).then(({data, errors}) => {
+      if(!errors && data.submitGenericAssessmentAnswers) {
+        setShowConclusions(true);
+      } else {
+        console.warn(errors || data); // eslint-disable-line no-console
+
+        // setTimeout(() => setSubmitAttempts((x) => x + 1), 2000);
```

Comment:
I think it would be good to add submit attempt logic similar to cognitive in case there's a random connection error or something